{"url":"https://api.github.com/repos/elastic/elasticsearch/issues/25305","repository_url":"https://api.github.com/repos/elastic/elasticsearch","labels_url":"https://api.github.com/repos/elastic/elasticsearch/issues/25305/labels{/name}","comments_url":"https://api.github.com/repos/elastic/elasticsearch/issues/25305/comments","events_url":"https://api.github.com/repos/elastic/elasticsearch/issues/25305/events","html_url":"https://github.com/elastic/elasticsearch/issues/25305","id":237078029,"node_id":"MDU6SXNzdWUyMzcwNzgwMjk=","number":25305,"title":"Error in cluster deployment: failed to send join request to master","user":{"login":"xiaoxing598","id":5615719,"node_id":"MDQ6VXNlcjU2MTU3MTk=","avatar_url":"https://avatars3.githubusercontent.com/u/5615719?v=4","gravatar_id":"","url":"https://api.github.com/users/xiaoxing598","html_url":"https://github.com/xiaoxing598","followers_url":"https://api.github.com/users/xiaoxing598/followers","following_url":"https://api.github.com/users/xiaoxing598/following{/other_user}","gists_url":"https://api.github.com/users/xiaoxing598/gists{/gist_id}","starred_url":"https://api.github.com/users/xiaoxing598/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/xiaoxing598/subscriptions","organizations_url":"https://api.github.com/users/xiaoxing598/orgs","repos_url":"https://api.github.com/users/xiaoxing598/repos","events_url":"https://api.github.com/users/xiaoxing598/events{/privacy}","received_events_url":"https://api.github.com/users/xiaoxing598/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":12,"created_at":"2017-06-20T02:49:55Z","updated_at":"2019-03-31T03:48:36Z","closed_at":"2017-06-20T04:27:22Z","author_association":"NONE","active_lock_reason":null,"body":"I have three virtual machines(192.168.245.128, 192.168.245.129, 192.168.245.130), respectively, in the above installed ES5.1.2, in the configuration of the cluster environment is encountered errors, the error in the three machines are similar.\r\n\r\nThis can be between the three machines can ping, you can telnet.\r\nThe following information is displayed for each machine.\r\n\r\n192.168.245.128\r\n\r\n```\r\n[2017-06-20T10:04:25,434][INFO ][o.e.t.TransportService   ] [node-1] publish_address {192.168.245.128:9300}, bound_addresses {192.168.245.128:9300}\r\n[2017-06-20T10:04:25,440][INFO ][o.e.b.BootstrapCheck     ] [node-1] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks\r\n[2017-06-20T10:04:33,811][INFO ][o.e.d.z.ZenDiscovery     ] [node-1] failed to send join request to master [{node-2}{X-m7gPTMQn2TsdlByavfEg}{S2ucttQDSXCqLjcyi7wjKA}{192.168.245.129}{192.168.245.129:9300}], reason [RemoteTransportException[[node-2][192.168.245.129:9300][internal:discovery/zen/join]]; nested: NotMasterException[Node [{node-2}{X-m7gPTMQn2TsdlByavfEg}{S2ucttQDSXCqLjcyi7wjKA}{192.168.245.129}{192.168.245.129:9300}] not master for join request]; ], tried [3] times\r\n```\r\n\r\n192.168.245.129\r\n\r\n```\r\n[2017-06-20T10:04:30,429][INFO ][o.e.t.TransportService   ] [node-2] publish_address {192.168.245.129:9300}, bound_addresses {192.168.245.129:9300}\r\n[2017-06-20T10:04:30,435][INFO ][o.e.b.BootstrapCheck     ] [node-2] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks\r\n[2017-06-20T10:04:33,813][INFO ][o.e.d.z.ZenDiscovery     ] [node-2] failed to send join request to master [{node-1}{X-m7gPTMQn2TsdlByavfEg}{YFmPk0dUTh-S1Ef9of5RTA}{192.168.245.128}{192.168.245.128:9300}], reason [RemoteTransportException[[node-1][192.168.245.128:9300][internal:discovery/zen/join]]; nested: NotMasterException[Node [{node-1}{X-m7gPTMQn2TsdlByavfEg}{YFmPk0dUTh-S1Ef9of5RTA}{192.168.245.128}{192.168.245.128:9300}] not master for join request]; ], tried [3] times\r\n```\r\n\r\n192.168.245.130\r\n\r\n```\r\n[2017-06-20T10:04:33,983][INFO ][o.e.t.TransportService   ] [node-3] publish_address {192.168.245.130:9300}, bound_addresses {192.168.245.130:9300}\r\n[2017-06-20T10:04:33,991][INFO ][o.e.b.BootstrapCheck     ] [node-3] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks\r\n[2017-06-20T10:04:37,354][INFO ][o.e.d.z.ZenDiscovery     ] [node-3] failed to send join request to master [{node-1}{X-m7gPTMQn2TsdlByavfEg}{YFmPk0dUTh-S1Ef9of5RTA}{192.168.245.128}{192.168.245.128:9300}], reason [RemoteTransportException[[node-1][192.168.245.128:9300][internal:discovery/zen/join]]; nested: NotMasterException[Node [{node-1}{X-m7gPTMQn2TsdlByavfEg}{YFmPk0dUTh-S1Ef9of5RTA}{192.168.245.128}{192.168.245.128:9300}] not master for join request]; ], tried [3] times\r\n```\r\n\r\nI chose a configuration file on a machine to show:\r\n\r\n```\r\ncluster.name: my-test\r\nnode.name: node-1\r\nnetwork.host: 192.168.245.128\r\ndiscovery.zen.ping.unicast.hosts: [\"192.168.245.129\", \"192.168.245.130\"]\r\ndiscovery.zen.minimum_master_nodes: 2\r\n```","closed_by":{"login":"jasontedor","id":4744941,"node_id":"MDQ6VXNlcjQ3NDQ5NDE=","avatar_url":"https://avatars3.githubusercontent.com/u/4744941?v=4","gravatar_id":"","url":"https://api.github.com/users/jasontedor","html_url":"https://github.com/jasontedor","followers_url":"https://api.github.com/users/jasontedor/followers","following_url":"https://api.github.com/users/jasontedor/following{/other_user}","gists_url":"https://api.github.com/users/jasontedor/gists{/gist_id}","starred_url":"https://api.github.com/users/jasontedor/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/jasontedor/subscriptions","organizations_url":"https://api.github.com/users/jasontedor/orgs","repos_url":"https://api.github.com/users/jasontedor/repos","events_url":"https://api.github.com/users/jasontedor/events{/privacy}","received_events_url":"https://api.github.com/users/jasontedor/received_events","type":"User","site_admin":false},"performed_via_github_app":null}